README.md

```diff
@@ -129,3 +61,232 @@ const stream = Kafka.KafkaConsumer.createReadStream(
+
+| Parameter                         | Type     |
+| --------------------------------- | -------- |
+| [type](#type)                     | `String` |
+| [client](#client)                 | `String` |
+| [bindingOptions](#bindingoptions) | `Object` |
+
+### type
+
+The type of service for which a binding is being
+requested. Currently the supported types are:
+
+- 'KAFKA'
+- 'POSTGRESQL'
+- 'REDIS'
+- 'MONGODB'
+- 'AMQP'
+- 'MYSQL'
+
+### client
+
+The package the application is using to connect
+to the service. kube-service-bindings is aware of a
+subset of possible packages. For those that it is aware
+of, it can map the service bindings into the form
+required by the client.
+
+Currently the following clients are recognized based on the supported types:
+
+- KAFKA
+  - node-rdkafka
+  - kafkajs
+- POSTGRESQL
+  - pg
+  - odbc
+- REDIS
+  - redis
+  - ioredis
+- MONGODB
+  - mongodb
+  - mongoose
+- AMQP
+  - rhea
+- MYSQL
+  - mysql
+  - mysql2
+  - odbc
+
+**(Deprecated)** If you don't specify a client, the object returned will
+be a direct map from the bindings, with the keys
+corresponding to the name of each file provided by the
+binding.
+
+#### Example on mongoDB client
+
+```javascript
+const serviceBindings = require('kube-service-bindings');
+const { MongoClient } = require('mongodb');
+
+const { url, connectionOptions } = serviceBindings.getBinding(
+  'MONGODB',
+  'mongodb'
+);
+
+const mongoClient = new MongoClient(url, connectionOptions);
+```
+
+### bindingOptions
+
+An object where provides control on parsing binding data.
+
+| Attribute                         | type      | default Value |
+| --------------------------------- | --------- | ------------- |
+| [id](#id)                         | `String`  | undefined     |
+| [removeUnmapped](#removeunmapped) | `Boolean` | true          |
+| [allwCopy](#allowcopy)            | `Boolean` | false         |
+| [bindingData](#bindingdata)       | `Object`  | undefined     |
+
+#### id
+
+Option id that is used to filter the available bindings in the search. Most useful for testing where you might have more than one binding of a given type available.
+
+#### removeUnmapped
+
+Fine grains parsed binding data to fit for the specified client, removing unnecessary properties from the input object.
+
+#### allowCopy
+
+Enables setting proper permissions for some of the binding data, where the system has not provided them correctly. It allows binding files content to be copied/stored in a new file and directory. This has to be enabled by the user in order to be aware of the security risk, as some files might include sensitive material. E.x. connecting to postgresql with odbc client, throws the following error for the tls.key file `permissions should be u=rw (0600) or less`.
+
+#### bindingData
+
+An object for passing binding data to kube-service-bindings. This is useful especially in local dev environtment or as a fallback in case of binding data are not available.
```

Comment:
```suggestion
An optional object for passing binding data to kube-service-bindings. This is useful especially in local dev environtment or as a fallback in case of binding data are not available.
```